ciao-launcher/main.go

```diff
@@ -237,15 +236,35 @@ func (client *agentClient) ErrorNotify(err ssntp.Error, frame *ssntp.Frame) {
 	glog.Infof("ERROR %d", err)
 }
 
-func (client *agentClient) installLauncherDeps() {
-	role := client.conn.Role()
-	osprepare.Bootstrap()
-	if role.IsNetAgent() {
-		osprepare.InstallDeps(launcherNetNodeDeps)
-	}
-	if role.IsAgent() {
-		osprepare.InstallDeps(launcherComputeNodeDeps)
+func (client *agentClient) installLauncherDeps(doneCh chan struct{}) {
+	ctx, cancelFunc := context.WithCancel(context.Background())
+
+	ch := make(chan error)
+	go func() {
+
+		osprepare.Bootstrap(ctx)
+
+		role := client.conn.Role()
+		if role.IsNetAgent() {
+			osprepare.InstallDeps(ctx, launcherNetNodeDeps)
+		}
+		if role.IsAgent() {
```

Comment:
If we have a launcher running in dual mode and we cancel the context during the first call to InstallDeps, we'll then try to call InstallDeps again, which will parse a file to determine the distro.  The easiest thing to do would be to return an error from InstallDeps if there is a cancel or a timeout.  There's a special error code for this I think.  I bet CommandContext returns it.  The alternative would be to select on the  context.Done() channel here.